### PR TITLE
prevents "bootstrap class path not set in conjunction with -source 8"

### PR DESCRIPTION
**Pull request by JerryShea**

### CODE REVIEW

File: third-party-bom/pom.xml [hunk 1]
```diff
@@ -644,6 +644,16 @@
                 <mockito.version>5.5.0</mockito.version>
             </properties>
         </profile>
+        <profile>
+            <!-- prevents "bootstrap class path not set in conjunction with -source 8" -->
+            <id>set-compiler-release</id>
```
Comment (JerryShea) [inline third-party-bom/pom.xml:649]: please see the profile above this ("java11") created by @peter-lawrey 

@peter-lawrey I don't understand the purpose of the above profile so need you to review this change. I have made my change next to yours, but please see below.

Why are we not affecting maven-compiler-plugin in a profile in [java-parent-pom](https://github.com/OpenHFT/OpenHFT/blob/c076135eed5f51f953fe2dcac2a8d993a9da0955/java-parent-pom/pom.xml#L80)?

Comment (peter-lawrey) [reply third-party-bom/pom.xml:649]: I suspect the line you are overriding should just be removed instead. Testing this locally.

Comment (yevgenp) [reply third-party-bom/pom.xml:649]: If we include JDK11 in this new profile then the existing above should be changed to `<jdk>(11,</jdk>` at least ?

Comment (JerryShea) [reply third-party-bom/pom.xml:649]: @peter-lawrey I tested this change with one of the demos and it worked like a charm. I am keen to get this merged

Comment (peter-lawrey) [reply third-party-bom/pom.xml:649]: This was solved in x.26 by setting -Xlint:-options

Comment (JerryShea) [reply third-party-bom/pom.xml:649]: Why have `-options` when instead we can do this and fix it properly, and we dont have to wait for x26?

Some projects have `-Xlint:all` and we should be aiming for this for all projects right?

**Comment by JerryShea** [comment]
@yevgenp your #222 PR supersedes this but basically just moves the compiler release stanzas to root-parent-pom?

**Comment by JerryShea** [comment]
#228 solves these problems with `-Xlint:-options`

**Pull request closed without merging**